code/modules/bitcoin/bitcoin_miner.dm

```diff
@@ -0,0 +1,132 @@
+/// This is the bitcoin miner!
+/// My code isn't going to be good. This is my first time!
+/// I will be editing the galactic materials market file
+
+/obj/machinery/bitcoin_miner
+	name = "bitcoin miner"
+	desc = "This machine mines bitcoins! To the moon!"
+	circuit = /obj/item/circuitboard/machine/bitcoin_miner
+	//req_access = list(ACCESS_CARGO) do i need to have this?
+	density = TRUE
+	icon = 'icons/obj/economy.dmi'
+	icon_state = "bit_miner"
+	base_icon_state = "bit_miner_inactive"
+	idle_power_usage = BASE_MACHINE_IDLE_CONSUMPTION
+	light_power = 0.7
+	light_range = MINIMUM_USEFUL_LIGHT_RANGE
+	var/static/list/money_insert = list(
+		/obj/item/holochip
+	)
+	var/static/list/min_insert = list(
+		/obj/item/organ/liver,
+		/obj/item/papercutter,
+		/obj/item/clothing/head/utility/bomb_hood,
+		/obj/item/clothing/suit/utility/bomb_suit,
+		/obj/item/clothing/gloves/color/fyellow,
+		/obj/item/trash/can,
+		/obj/item/food/deadmouse,
+		/obj/item/kitchen/rollingpin,
+		/obj/item/gun/energy/laser/practice,
+		/obj/item/phone,
+		///obj/item/book/manual/wiki/securityspacelaw, it errors out for some reason
+		/obj/item/dice/d8
+	)
+	var/static/list/max_insert = list(
+		/obj/item/clothing/head/bio_hood,
+		/obj/item/toner/large,
+		/obj/item/flashlight/flare,
+		/obj/item/wallframe/camera,
+		/obj/item/clothing/mask/breath,
+		/obj/item/pen/fourcolor,
+		/obj/item/clothing/suit/hazardvest,
+		/obj/item/storage/briefcase/secure,
+		/obj/item/gavelhammer,
+		/obj/item/clothing/neck/tie/black
+	)
+	var/upgrade_state = 1
+	var/bitcoin_balance = 0
+	var/min = 1
+	var/max = 2
+	var/got = 0
+	var/perc = 20
+
+/obj/machinery/bitcoin_miner/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_unfasten_wrench(user, tool, time = 1.5 SECONDS) == SUCCESSFUL_UNFASTEN)
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/materials_market/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_deconstruction_screwdriver(user, base_icon_state, base_icon_state, tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/bitcoin_miner/update_icon_state()
+	if(!powered() && icon_state == "bit_miner")
+		icon_state = base_icon_state
+		return ..()
+	if(powered() && icon_state == "bit_miner_inactive")
+		icon_state = "bit_miner"
+		return ..()
+
+/obj/machinery/bitcoin_miner/crowbar_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_deconstruction_crowbar(tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/bitcoin_miner/attackby(obj/item/O, mob/user, params)
+	if(is_type_in_list(O, money_insert))
+		var/value = O.get_item_credit_value()

```

Comment:
Avoid one letter variables and use typecaches instead of lists.
Also, have you even tested this? You're looking for an object inside of a list of types, this will never work.

Reply:
yeah i did test it. It works pretty fine. But I'm remaking it currently

Reply:
it does not function on any subtypes of the items in the list